### PR TITLE
update timingscripts

### PR DESCRIPTION
**Pull request by kdelange**

### CODE REVIEW

File: bin/pipelineTiming.sh [hunk 1]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
-            continue
-        fi
-    fi
-
-    echo -e "The complete pipeline is finished for project ${project}" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
-    log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "The complete pipeline is finished for project ${project}"
-    
-    ssh ${sourceServerFQDN} "mv ${SCR_ROOT_DIR}/logs/Timestamp/${project}.csv ${SCR_ROOT_DIR}/logs/Timestamp/archive/"
-    mv "${projectStartDir}/${project}.csv" "${projectStartDir}/archive/"
+	if [[ ! -e "${projectSheet}" ]]
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: Kan eruit

File: bin/pipelineTiming.sh [hunk 2]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
 for projectSheet in $(ls "${projectStartDir}/"*".csv")
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: check if $(ls "${projectStartDir}/"*".csv") exists and then do for loop

File: bin/pipelineTiming.sh [hunk 3]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
-    echo -e "The complete pipeline is finished for project ${project}" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
-    log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "The complete pipeline is finished for project ${project}"
-    
-    ssh ${sourceServerFQDN} "mv ${SCR_ROOT_DIR}/logs/Timestamp/${project}.csv ${SCR_ROOT_DIR}/logs/Timestamp/archive/"
-    mv "${projectStartDir}/${project}.csv" "${projectStartDir}/archive/"
+	if [[ ! -e "${projectSheet}" ]]
+	then
+		continue
+	fi
+	project=$(basename "${projectSheet}" .csv)
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: no space between "${projectSheet}" and .csv?

File: bin/pipelineTiming.sh [hunk 4]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+
+	touch "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+	echo -e "moment of checking run time: $(date)\nproject: ${project}\n" > "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+	log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Using project: ${project}"
+
+	if [[ ! -d "${logsDir}/${project}/" ]]
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: Add line 218-220 in the else of this for loop.

File: bin/pipelineTiming.sh [hunk 5]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+	then
+		continue
+	fi
+
+	## step 1, checks the generatedScirps
+	if [ -e "${logsDir}/${project}/${run}.generateScripts.finished" ]
+	then
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: delete tab after "then"

File: bin/pipelineTiming.sh [hunk 6]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
-    
-    ssh ${sourceServerFQDN} "mv ${SCR_ROOT_DIR}/logs/Timestamp/${project}.csv ${SCR_ROOT_DIR}/logs/Timestamp/archive/"
-    mv "${projectStartDir}/${project}.csv" "${projectStartDir}/archive/"
+	if [[ ! -e "${projectSheet}" ]]
+	then
+		continue
+	fi
+	project=$(basename "${projectSheet}" .csv)
+
+	touch "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 7]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+	echo -e "moment of checking run time: $(date)\nproject: ${project}\n" > "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+	log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Using project: ${project}"
+
+	if [[ ! -d "${logsDir}/${project}/" ]]
+	then
+		continue
+	fi
+
+	## step 1, checks the generatedScirps
+	if [ -e "${logsDir}/${project}/${run}.generateScripts.finished" ]
```
Comment (RoanKanninga) [inline bin/pipelineTiming.sh]: test

File: bin/pipelineTiming.sh [hunk 8]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+	then	
+		echo -e "generatedScripts finished for project ${project}" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 9]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+		touch "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.finished"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 10]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "generateScripts finished for project ${project}"
+	else
+		timeStampGeneratedScripts=$(find "${logsDir}/${project}/" -type f -mmin +240 -iname "${run}.generateScripts.started")
+		if [[ -z "${timeStampGeneratedScripts}" ]]
+		then
+			echo -e "generateScipts has not started yet or is running" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 11]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "generateScipts has not started yet or is running for project ${project}"
+			continue
+		else
+			echo -e "generateScripts.started file is OLDER than 4 hour.\n" \
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: typo: hours

File: bin/pipelineTiming.sh [hunk 12]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+			"time ${run}.generateScripts.started was last modified:" \
+			$(stat -c %y "${TMP_ROOT_DIR}/logs/${project}/${run}.generateScripts.started") >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+		
+			touch "${TMP_ROOT_DIR}/logs/${project}/${run}.generateScriptsTiming.failed"
+			echo -e "Dear HPC helpdesk,\n\nPlease check if there is something wrong with the pipeline.\nThe generatedScripts step for project ${project} is not finished after 5h.\n\nKind regards\nHPC" > "${TMP_ROOT_DIR}/logs/${project}/${run}.generateScriptsTiming.failed"
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "generateScripts.started file is OLDER than 5 hour for project ${project}"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: 4 hours

File: bin/pipelineTiming.sh [hunk 13]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+			continue
+		fi
+	fi
+
+	## step 2, checks the pipeline
+	if [ -e "${logsDir}/${project}/${run}.pipeline.finished" ]
+	then
+		echo "pipeline is finished for project ${project}" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 14]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+		touch "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.finished"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 15]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+			echo -e "pipeline is has not started yet or is still running" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "pipeline has not started yet or is running for project ${project}"
+			continue
+		else
+			echo -e "pipeline.started file is OLDER than 10 hour.\n" \
+			"time ${run}.pipeline.started was last modified:" \
+			$(stat -c %y "${TMP_ROOT_DIR}/logs/${project}/${run}.pipeline.started") >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+			touch "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.failed"
+			echo -e "Dear HPC helpdesk,\n\nPlease check if there is something wrong with the pipeline.\nThe pipeline for project ${project} is not finished after 6h.\n\nKind regards\nHPC" > "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.failed"
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "pipeline.started file is OLDER than 6 hour for project ${project}"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: 10 hours

File: bin/pipelineTiming.sh [hunk 16]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+			continue
+		fi
+	fi 
+
+	## step 3, check the copyProjectDataToPrm
+	if [ -e "${logsDir}/${project}/${run}.copyProjectDataToPrm.finished" ]
+	then
+		echo -e "copyProjectDataToPrm is finished for project ${project}" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 17]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+		touch "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrmTiming.finished"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 18]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+			echo -e "copyProjectDataToPrm has not started jet, or is running" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "copyProjectDataToPrm has not started yet or is running for project ${project}"
+			continue
+		else
+			echo -e "copyProjectDataToPrm.started file is OLDER than 4 hour.\n" \ 
+			"time ${run}.copyProjectDataToPrm.started was last modified:\n" \
+			$(stat -c %y "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrm.started") >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+			touch "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrmTiming.failed"
+			echo -e "Dear HPC helpdesk,\n\nPlease check if there is something wrong with the pipeline.\nThe copyProjectDataToPrm has started but is not finished after 6h for project ${project}.\n\nKind regards\nHPC" > "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrmTiming.failed"
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "copyProjectDataToPrm.started file OLDER than 5 hour for project ${project}"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: 4 hours

File: bin/pipelineTiming.sh [hunk 19]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+	if [ -e "${logsDir}/${project}/${run}.copyProjectDataToPrm.finished" ]
+	then
+		echo -e "copyProjectDataToPrm is finished for project ${project}" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+		touch "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrmTiming.finished"
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "copyProjectDataToPrm finished for project ${project}"
+	else
+		timeStampCopyProjectDataToPrm=$(find "${logsDir}/${project}/" -type f -mmin +240 -iname "${run}.copyProjectDataToPrm.started")
+		if [[ -z "${timeStampCopyProjectDataToPrm}" ]]
+		then
+			echo -e "copyProjectDataToPrm has not started jet, or is running" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh [hunk 20]
```diff
@@ -209,99 +209,99 @@ rsync -av ${sourceServerFQDN}:"${SCR_ROOT_DIR}/logs/Timestamp/*.csv" "${projectS
...
+			"time ${run}.copyProjectDataToPrm.started was last modified:\n" \
+			$(stat -c %y "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrm.started") >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
+			touch "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrmTiming.failed"
+			echo -e "Dear HPC helpdesk,\n\nPlease check if there is something wrong with the pipeline.\nThe copyProjectDataToPrm has started but is not finished after 6h for project ${project}.\n\nKind regards\nHPC" > "${TMP_ROOT_DIR}/logs/${project}/${run}.copyProjectDataToPrmTiming.failed"
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "copyProjectDataToPrm.started file OLDER than 5 hour for project ${project}"
+			continue
+		fi
+	fi
+
+	echo -e "The complete pipeline is finished for project ${project}" >> "${TMP_ROOT_DIR}/logs/${project}/${run}.${SCRIPT_NAME}.log"
```
Comment (marieke-bijlsma) [inline bin/pipelineTiming.sh]: You can use ${logsDir} instead of ${TMP_ROOT_DIR}/logs/

File: bin/pipelineTiming.sh (hunk 3 above)
Comment (kdelange) [reply bin/pipelineTiming.sh]: nee, projectsheet heeft als excetie .csv. met die .csv erachter haal je dat er af.

**Pull request merged**